internal/controller/pdb_controller.go

```diff
@@ -0,0 +1,131 @@
+	return r.handleDeploymentCreation(ctx, &deployment)
+}
+
+// handleDeploymentCreation creates a PodDisruptionBudget when a Deployment is created or updated.
+func (r *DeploymentPDBReconciler) handleDeploymentCreation(ctx context.Context, deployment *v1.Deployment) (reconcile.Result, error) {
+	log := log.FromContext(ctx)
+	// Check if PDB already exists for this Deployment
+	pdb := &policyv1.PodDisruptionBudget{}
+	err := r.Get(ctx, client.ObjectKey{
+		Namespace: deployment.Namespace,
+		Name:      deployment.Name + "-pdb",
+	}, pdb)
+
+	if err == nil {
+		// PDB already exists, nothing to do
+		log.Info("PodDisruptionBudget already exists", "namespace", deployment.Namespace, "name", deployment.Name)
+		return reconcile.Result{}, nil
+	} else if client.IgnoreNotFound(err) != nil {
+		// If there was an error fetching the PDB (other than NotFound), log and return
+		log.Error(err, "unable to fetch PDB")
+		return ctrl.Result{}, err
+	}
+
+	// Create a new PDB for the Deployment
+	pdb = &policyv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deployment.Name + "-pdb",
+			Namespace: deployment.Namespace,
+		},
+		Spec: policyv1.PodDisruptionBudgetSpec{
+			MinAvailable: &intstr.IntOrString{IntVal: 1},
+			Selector:     &metav1.LabelSelector{MatchLabels: deployment.Spec.Selector.MatchLabels},
+		},
+	}
+
+	log.Info("Creating PodDisruptionBudget", "namespace", pdb.Namespace, "name", pdb.Name)
+	if err := r.Create(ctx, pdb); err != nil {
+		log.Error(err, "unable to create PDB")
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// handleDeploymentDeletion deletes the associated PDB when the Deployment is deleted
+func (r *DeploymentPDBReconciler) handleDeploymentDeletion(ctx context.Context, deployment *v1.Deployment) (reconcile.Result, error) {
+	log := log.FromContext(ctx)
+
+	// Try to get the associated PDB
+	pdb := &policyv1.PodDisruptionBudget{}
+	err := r.Get(ctx, client.ObjectKey{
+		Namespace: deployment.Namespace,
+		Name:      deployment.Name + "-pdb",
+	}, pdb)
+
+	if err != nil {
+		// If there's no PDB or it can't be fetched, just return
+		log.Info("PodDisruptionBudget does not exist or error fetching", "namespace", deployment.Namespace, "name", deployment.Name)
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// If the PDB exists, delete it
+	log.Info("Deleting PodDisruptionBudget", "namespace", pdb.Namespace, "name", pdb.Name)
+	if err := r.Delete(ctx, pdb); err != nil {
+		log.Error(err, "unable to delete PDB")
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *DeploymentPDBReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Set up the controller to watch Deployments and trigger the reconcile function
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1.Deployment{}).
+		WithEventFilter(predicate.Funcs{
```

Comment:
if you don't specify is the default true? either set updatefunc to return false or don't have this filter at all?